### PR TITLE
Generate runtime API reference documentation

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
Updates the API reference generator script to support the APIs defined in the `runtime` repository. Also updates the HTML of the API reference page to place the APIs in their own section at the top under `Common`.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/scripts/gen_reference.py [hunk 1]
```diff
@@ -144,6 +146,7 @@ def convert_crd_to_resource(crd: Dict, service) -> Resource:
     return Resource(
         name=crd['spec']['names']['kind'],
         service=service,
+        service_package=f"{service}-controller/apis" if service_package == "" else service_package,
```
Comment (jaypipes) [inline docs/scripts/gen_reference.py:149]: this seems backward to me, no?

Comment (RedbackThomson) [reply docs/scripts/gen_reference.py:149]: The Pythonic ternary operator just feels backwards to me in general. But this is correct. `{true_statement} if {condition} else {false_statement}`

Comment (RedbackThomson) [reply docs/scripts/gen_reference.py:149]: Hmm let me just rewrite this as a Python default argument value

**Review by jaypipes — APPROVED** [review]
great work on this, @RedbackThomson

**Pull request merged**